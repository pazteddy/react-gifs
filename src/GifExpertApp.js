import React, { useState } from 'react';
import { AddCategory } from './components/AddCategory';
import { GifGrid } from './components/GifGrid';
const GifExpertApp = ({ defaultCategorias =[]})=>{

  // const categorias=['One Punch','Heroe Academia','Dragon Ball'];
  // const [categorias, setCategorias] = useState(['Heroe Academia']);
  const [categorias, setCategorias] = useState( defaultCategorias );
  // const handleAdd = () =>{
  //   //setCategorias([...categorias,'Garfield']);
  //   setCategorias( cats => [...cats,'Garfield']);
  // }

  return(
    <>
    <h2>GifExpertapp</h2>
    <AddCategory setCategorias={setCategorias}/>
    <hr/>
    {/* <button onClick= {handleAdd}>Agregar</button> */}
    <ol>
      { 
        categorias.map( categoria =>
          <GifGrid 
          key={categoria}
          categoria = {categoria}/>
        )
      }
    </ol>
    </>
  )
}
export default GifExpertApp;
//Snippet rafc
// import React from 'react'

// export const GifExpertApp = () => {
//   return (
//     <div>
      
//     </div>
//   )
// }
