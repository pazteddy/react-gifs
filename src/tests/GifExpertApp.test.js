import React from 'react';
import { shallow } from 'enzyme';
import GifExpertApp from '../GifExpertApp';

describe('Prueba de funcionalidad <GifExpertApp/>', () => {
  test('Debe mostrarse correctamente <GifExpertApp/>', () => {
    const wrapper= shallow(<GifExpertApp/>);
    expect( wrapper ).toMatchSnapshot();
  });

  test(' Debe mostrar una lista de categorias ', () => {
    const categorias = ['Heores Academia', 'Garfield'];
    const wrapper= shallow(<GifExpertApp defaultCategorias={ categorias }/>);
    expect(wrapper).toMatchSnapshot();
    expect( wrapper.find('GifGrid').length).toBe(categorias.length);

  });
  

});
