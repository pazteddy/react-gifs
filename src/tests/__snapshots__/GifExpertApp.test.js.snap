// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prueba de funcionalidad <GifExpertApp/>  Debe mostrar una lista de categorias  1`] = `
<Fragment>
  <h2>
    GifExpertapp
  </h2>
  <AddCategory
    setCategorias={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      categoria="Heores Academia"
      key="Heores Academia"
    />
    <GifGrid
      categoria="Garfield"
      key="Garfield"
    />
  </ol>
</Fragment>
`;

exports[`Prueba de funcionalidad <GifExpertApp/> Debe mostrarse correctamente <GifExpertApp/> 1`] = `
<Fragment>
  <h2>
    GifExpertapp
  </h2>
  <AddCategory
    setCategorias={[Function]}
  />
  <hr />
  <ol />
</Fragment>
`;
